src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComponentResourceManager.cs

```diff
@@ -149,10 +149,10 @@ public virtual void ApplyResources(object value, string objectName, CultureInfo
                     }
                 }
 
-                // Character after objectName.Length should be a ".", or else we should continue.
+                // Character after objectName.Length should be a "." or a '-', or else we should continue.
```

Comment:
Unlikely because the code will just continue to the rest of the loop and will just continue as it used to be. The only breaking scenario if you have the resources like "Object-Property" and at the same time we have code like Object.Property. considering the usage of this class and considering the probability of this case I doubt this case exists at all.